### PR TITLE
Add SherdNote api with validation - MEDT-3500

### PR DESCRIPTION
**Pull request by nikolas**
Using django-rest-framework.

### CODE REVIEW

File: mediathread/djangosherd/apiviews.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from mediathread.djangosherd.serializers import SherdNoteSerializer
+from mediathread.sequence.apiviews import CsrfExemptSessionAuthentication
+
+
+class SherdNoteCreate(CreateAPIView):
+    authentication_classes = [CsrfExemptSessionAuthentication]
+    permission_classes = [IsAuthenticated]
```
Comment (sdreher) [inline mediathread/djangosherd/apiviews.py]: Additional permissions are needed to ensure the user can only create an annotation on an asset that is visible to them. We need a check that the asset is within a course where the user is a member. And, that the asset can be edited based on the course permissions.

Comment (nikolas) [reply mediathread/djangosherd/apiviews.py:12]: Thanks for listing these out! I will add these tomorrow.

File: mediathread/djangosherd/permissions.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            raise PermissionDenied
+
+        return True
+
+
+class AssetInMyCourse(BasePermission):
+    message = 'You can only update assets in your courses.'
+
+    def has_object_permission(self, request, view, obj):
+        # Am I the owner of this asset?
```
Comment (sdreher) [inline mediathread/djangosherd/permissions.py:31]: This first check duplicates `AssetIsVisible`.  I don't think it is necessary here.

File: mediathread/djangosherd/permissions.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        if obj.author == request.user:
+            return True
+
+        if cached_course_is_faculty(obj.course, request.user):
+            return True
+
+        if not all_items_are_visible(request.course):
```
Comment (sdreher) [inline mediathread/djangosherd/permissions.py:21]: ...should this check use `obj.course` for consistency?

File: mediathread/djangosherd/serializers.py [hunk 4]
```diff
@@ -15,11 +15,73 @@ class Meta:
...
+
+        if not range1:
+            raise serializers.ValidationError({
+                'range1': 'This field is required.'
+            })
+
+        if not range2:
+            raise serializers.ValidationError({
+                'range2': 'This field is required.'
+            })
```
Comment (sdreher) [inline mediathread/djangosherd/serializers.py:74]: Is a check needed here to ensure range1 < range2?

Comment (nikolas) [reply mediathread/djangosherd/serializers.py:74]: Yes, I just added that.

**Pull request merged**